records.py

```diff
@@ -328,6 +328,7 @@ def transaction(self):
             tx.commit()
         except:
             tx.rollback()
+            raise
         finally:
             conn.close()
```

Comment:
Does the finally block run and the connection close if you reraise within the except block?

Reply:
Yep, finally is always run, whatever happens inside the rest of the try-except

See the second example here: https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions

Reply:
Thanks for contribution @brendanator, I'm wondering what do you think about raising error with semantic description?

Reply:
As we don't know what type of exception has been originally thrown I don't think this is a good idea. It could be an sql exception (like duplicate key) or it could be a custom exception raised in the code. By just re-raising it directly the surrounding code can handle it as needed